src/applications/mhv-landing-page/utilities/data/resolveLinks.js

```diff
@@ -11,14 +14,31 @@ export const toggleLink = (link, featureToggles = {}) => {
     ariaLabel,
   } = link;
   const showNewLink =
-    (!!toggle && (hasOwn(featureToggles, toggle) && featureToggles[toggle])) ||
-    (!oldText && !oldHref);
+    isToggled(toggle, featureToggles) || (!oldText && !oldHref);
+
   const href = showNewLink ? newHref : oldHref;
   const text = showNewLink ? newText : oldText;
-  return { href, text, key: toggle, ariaLabel };
+  const isExternal = !!link.isExternal;
+  return { href, text, key: toggle, ariaLabel, isExternal };
 };
 
-const resolveLinks = (links, featureToggles) =>
-  links.map(l => toggleLink(l, featureToggles)).filter(l => !!l.href);
+// Removes links that (all must be true):
+// 1. Are not toggled (feature flag disabled)
+// 2. Have no `oldText` or `oldHref`
+// 3. Should be "hard toggled" (completely disabled)
+//     e.g. in the case of a brand-new link with no `oldText`
+//     or in the event the link should be removed quickly and completely
+const filterOutHideableLinks = (links, featureToggles) => {
+  const shouldHideFilter = link =>
+    !isToggled(link.toggle, featureToggles) && link.hardToggle;
+  return links.filter(l => !shouldHideFilter(l));
+};
```

Comment:
This is the new stuff, and basically supports a `hardToggle` flag, which tells if a link should be completely invisible to the user if the feature flag is disabled.

I would love to hear more elegant options to this!

Reply:
This is relatively minor as what you have does work, however it would be possible to do the following:

`const filterOutHideableLinks = (links, featureToggles) => {
  return links.filter(link => !(
    !isToggled(link.toggle, featureToggles) && link.hardToggle
  ));
};`

Reply:
Good catch, thanks for improving the readability of this!